istioctl/pkg/injector/injector-list.go

```diff
@@ -249,6 +249,9 @@ func getMatchingNamespaces(hook *admitv1.MutatingWebhookConfiguration, namespace
 	retval := make([]corev1.Namespace, 0, len(namespaces))
 	seen := sets.String{}
 	for _, webhook := range hook.Webhooks {
+		if webhook.Name != "rev.namespace.sidecar-injector.istio.io" {
+			continue
```

Comment:
Because `getMatchingNamespaces` function will be called multiple times depending on the number of mutatingwebhookconfigurations
https://github.com/istio/istio/blob/master/istioctl/pkg/injector/injector-list.go#L201-L203

Reply:
In fact, the main problem is not duplicate data, but the previous logic of webhook matching too many namespaces: https://github.com/istio/istio/pull/53583#discussion_r1861629337

Reply:
I think filtering the object webhooks is the right approach here, as they will also match the namespace that has no istio related label. However `rev.namespace.sidecar-injector.istio.io` only handles the revision object, and we still have default webhook like `object.sidecar-injector.istio.io`.

Reply:
Thanks for the review. Here, the default webook should use `namespace.sidecar-injector.istio.io`?
```
$ kubectl get mutatingwebhookconfigurations.admissionregistration.k8s.io  istio-sidecar-injector -oyaml |grep injector.istio.io
  name: rev.namespace.sidecar-injector.istio.io
  name: rev.object.sidecar-injector.istio.io
  name: namespace.sidecar-injector.istio.io
  name: object.sidecar-injector.istio.io
```

Reply:
Both namespace webhooks can be used for matching, you can see https://github.com/istio/istio/blob/master/manifests/charts/default/templates/mutatingwebhook.yaml for what they are doing.